### DISCUSSION TITLE
ddt4all obdII elm327 Bluetooth com 3/4 access denied

### DISCUSSION THREAD

**Discussion by letrasluis2** [cedricp/ddt4all#1068] [Q&A] [answered]
Hi, 

I cannot use DDT4ALL versions 2.1.1 /3.0.1 /5.5.0 with Win11x64 and ELM327 Bluetooth - permission error for COM3 and COM4.

I used version 2-1-1 and 3.0.1 and 5.5.0 and i have checked https://github.com/cedricp/ddt4all/issues/778 and i have confirmed my py files are updated with same code: "-- self.screenlogfile = open("./logs/screens.txt", "at")
++self.screenlogfile = open("./logs/screens.txt", "at", encoding='utf-8')"

I used version 3.0.1 from https://github.com/cedricp/ddt4all/releases/download/v3.0.1/DTT4ALL-Windows-Installer-v3.0.1_Stable-x86-vcredist.exe with updated ECU db with no sucess.

I have used different ddt4all versions in order to solve connectivity but there is no joy.

I have reset pc, and Bluetooth devices.

I have seen this errors in some Internet posts but than the discussion summary is lost so I have no idea how can resolve
![20240330_101731](https://github.com/cedricp/ddt4all/assets/165518226/35709f42-8a7b-422f-987d-2096956c8676)

![20240330_105200](https://github.com/cedricp/ddt4all/assets/165518226/2e9f75e0-33e2-407d-9b80-058c0e70e626)
![20240330_104807](https://github.com/cedricp/ddt4all/assets/165518226/6c03a40f-4e4b-48b3-a18a-65be83968c2f)

![20240330_101743](https://github.com/cedricp/ddt4all/assets/165518226/0745a874-d958-4a5f-805d-24e196e68aab)

![IMG-20240331-WA0006](https://github.com/cedricp/ddt4all/assets/165518226/5fda2700-aa06-478a-9318-c494b88b3626)
![IMG-20240331-WA0010](https://github.com/cedricp/ddt4all/assets/165518226/c1d220f6-4f34-4116-820f-85f6b583fc62)
![IMG-20240331-WA0008](https://github.com/cedricp/ddt4all/assets/165518226/230d5897-0950-4bec-b64e-72c1966916c7)
![IMG-20240331-WA0009](https://github.com/cedricp/ddt4all/assets/165518226/f7225ca4-b617-4a79-8424-ba05358caefe)
![IMG-20240331-WA0007](https://github.com/cedricp/ddt4all/assets/165518226/d28d2446-7835-4534-b1a8-f5ae44659906)
![IMG-20240331-WA0005](https://github.com/cedricp/ddt4all/assets/165518226/67b2151f-3ced-4149-8310-036d4ba8a62b)

I enjoyed the efforts from everyone in other posts.

thans in advance

**Comment by KarelSvo** [comment]
email sent

**Comment by letrasluis2** [comment]
> email sent
meanwhile I arrange usb device, and I have tested it with your sw: ELM not responding


![20240402_120332](https://github.com/cedricp/ddt4all/assets/165518226/8b0e7067-517f-407f-8a19-a80ccee58ac6)
![IMG-20240402-WA0006](https://github.com/cedricp/ddt4all/assets/165518226/1e42241f-c645-4a21-a932-761782dc77ed)
![IMG-20240402-WA0001](https://github.com/cedricp/ddt4all/assets/165518226/e7e3c417-50db-4526-a143-5907799539a5)
![IMG-20240402-WA0003](https://github.com/cedricp/ddt4all/assets/165518226/f38a39bf-1dfd-4a71-b379-3114e70a821f)

**Comment by letrasluis2** [comment]
> email sent

with elm Bluetooth COM 3

![20240402_120148](https://github.com/cedricp/ddt4all/assets/165518226/aaa4b4b3-5970-4b9d-9667-9fa6bf0ae7bf)

COM 4 overflow 

![IMG-20240402-WA0002](https://github.com/cedricp/ddt4all/assets/165518226/8c452f00-0626-443c-8ed7-bbe609b5f918)

![IMG-20240402-WA0004](https://github.com/cedricp/ddt4all/assets/165518226/e4ac2bc8-17c8-4ccf-a71e-37f694afdf7f)

**Comment by letrasluis2** [comment]
> email sent

 with elm Bluetooth COM 4

![20240402_120148](https://github.com/cedricp/ddt4all/assets/165518226/9b58120c-0445-4d4b-b479-d0a842718e22)


com 3 show some results and com 4 overflow in connection test...

![IMG-20240402-WA0008(1)](https://github.com/cedricp/ddt4all/assets/165518226/53b383c8-475f-4fea-9793-e28c052153da)
![IMG-20240402-WA0007](https://github.com/cedricp/ddt4all/assets/165518226/4e094a37-20e6-48fa-b783-d9880005c7bf)
![IMG-20240402-WA0005](https://github.com/cedricp/ddt4all/assets/165518226/f02fe36f-54c2-4954-81e2-ced196f88e54)


full log:

Opening ELM
Checking port speed: 115200
Start COM speed:  115200

Cannot determine OBDLink version.
OBDLink performance may be decreased.
Press any key to continue...

v1.0   ATZ                    [TIMEOUT]
v1.0   ATWS                   [TIMEOUT]
v1.0   ATE1                   [TIMEOUT]
v1.0   ATL1                     [OK]
v1.0   ATL0                     [OK]
v1.0   ATI                      [OK]
v1.0   ATRV                     [OK]
v1.1   ATPPS                  [FAIL]
v1.3   ATS1                     [OK]
v1.3   ATS0                     [OK]
v1.0   ATCAF0                   [OK]
v1.0   ATCFC0                   [OK]
v1.0   ATH0                     [OK]
v1.0   ATR1                     [OK]
v1.2   ATKW0                    [OK]
v1.3   ATD1                     [OK]
v1.3   ATD0                     [OK]
v1.0   ATAL                     [OK]
V1.0   ATSP 3                   [OK]
V1.0   ATSP 4                   [OK]
v1.2   ATIIA 01                 [OK]
v1.4   ATSI                   [FAIL]
V1.0   ATSP 5                   [OK]
v1.4   ATFI                   [FAIL]
V1.0   ATSP 6                   [OK]
V1.0   ATSP 7                   [OK]
V1.0   ATSP 8                   [OK]
v1.0   ATPC                     [OK]
v1.0   ATSH 012                 [OK]
v1.0   ATSW 96                  [OK]
v1.0   ATSW 34                  [OK]
v1.0   ATWM 817AF13E            [OK]
v1.0   ATIB 10                  [OK]
v1.0   ATST FF                  [OK]
v1.3   ATCRA 012                [OK]
v1.3   ATCRA 01234567           [OK]
v1.1   ATFC SH 012              [OK]
v1.1   ATFC SH 00112233         [OK]
v1.1   ATFC SD 300000           [OK]
v1.1   ATFC SM 1                [OK]
v1.2   ATAT0                    [OK]
v1.2   ATAT1                    [OK]
v1.0   ATCM 123                 [OK]
v1.0   ATCM 12345678            [OK]
v1.0   ATCF 123                 [OK]
v1.0   ATCF 12345678            [OK]
v1.0   AT@1                     [OK]
STN TEST 1  STI               [FAIL]
STN TEST 2  STDI              [FAIL]
****************************************
*       RESETTING ELM
****************************************
*
*       ERRORS STATISTIC
*
* FRAME      =  0
* BUFFERFULL =  0
* QUESTION   =  5
* NODATA     =  0
* TIMEOUT    =  5
* RX         =  0
* CAN        =  0
*
*       RESPONSE TIME (Average)
*
* response_time    =  0.000
*
****************************************



Uncompatible adapter on ARM core
 pyren would not work with it





Result: 41 from 49
 Max version:v1.0








C:\Program Files (x86)\PyRen3>

**Comment by KarelSvo** [comment]
The recommendation is, to replace the unsuitable adapters with a vLinker FS USB. v1.1 ATPPS [FAIL] shows, it is a very bad copy.

**Comment by letrasluis2** [comment]
> The recommendation is, to replace the unsuitable adapters with a vLinker FS USB. v1.1 ATPPS [FAIL] shows, it is a very bad copy.

like this ?

OBD2 Car Diagnostic Scanner, Ferramentas de Interface, Ford FORScan, Mazda VLinker FS, USB FS, ELM327, ELM 327, OBD 2, OBDII, HS, MS-CAN
https://a.aliexpress.com/_EQYJOLf

**Comment by KarelSvo** [comment]
yes

**Comment by letrasluis2** [comment, marked as answer]
> yes
problem with ELM is resolved, i have a new with quality.
Will open another post for the issue i have with my rlink radio. thanks
